anime/views.py

```diff
@@ -113,3 +119,36 @@ def create_anime():
     db.session.commit()
 
     return anime_schema.dump(schema_dict), 201
+
+
+@animes_bp.route('/api/v1/register', methods=['POST'])
+def register():
+    username = request.json.get("username")
+    password = request.json.get("password")
+
+    user = User.query.filter_by(username=username).one_or_none()
+
+    if user is not None:
+        return jsonify(message='username exists'), 400
+
+    hashed_password = generate_password_hash(password)
+
+    user = User(username=username, password=hashed_password)
+    db.session.add(user)
+    db.session.commit()
+    return jsonify(message='user created'), 201
+
+
+@animes_bp.route('/api/v1/login', methods=['POST'])
+def login():
+    username = request.json.get("username")
+    password = request.json.get("password")
+
```

Comment:
Same here. We should check the user schema before checking types before looking for the user in the database.